src/model/cloud-runner/tests/e2e/cloud-runner-end2end-retaining.test.ts

```
@@ -47,6 +48,33 @@ describe('Cloud Runner Retain Workspace', () => {
+      if (CloudRunnerOptions.providerStrategy === 'k8s') {
+        try {
+          CloudRunnerLogger.log('Cleaning up k3d node between builds (preserving Unity image)...');
+          const K3D_NODE_CONTAINERS = ['k3d-unity-builder-agent-0', 'k3d-unity-builder-server-0'];
+          for (const NODE of K3D_NODE_CONTAINERS) {
+            // Remove stopped containers but keep images
+            await CloudRunnerSystem.Run(
+              `docker exec ${NODE} sh -c "crictl rm --all 2>/dev/null || true" || true`,
+              true,
+              true,
+            );
+            // Only remove specific known system images, preserve Unity and everything else
```

ESLint

[error] Expected line before comment.

(lines-around-comment)